pkg/controller/migmigration/task.go

```diff
@@ -749,9 +774,8 @@ func (t *Task) anyFlags(step Step) bool {
 }
 
 // Phase fail.
-func (t *Task) fail(nextPhase string, reasons []string) {
+func (t *Task) fail(nextPhase string, reasons ...string) {
```

Comment:
This signature was changed to support the _exception_ use case which seems undesirable and resulted in a lot of unrelated code changes.

Reply:
It's not obvious to me exactly what's going on here. I'm not seeing any obvious places where reasons needs the change, and the method in question isn't doing anything different with reasons than before.